permission/permission.go

```diff
@@ -855,3 +868,104 @@ func (p *PermissionCtrl) manageRolePermissions() error {
+	permAcctInterface := &pbind.AcctManagerSession{
+		Contract: p.permAcct,
+		CallOpts: bind.CallOpts{
+			Pending: true,
+		},
+	}
+	account, orgId, roleId, status, isAdmin, _ := permAcctInterface.GetAccountDetails(acctId)
+	if status.Int64() == 0 {
+		return nil
+	}
+	return &types.AccountInfo{AcctId: account, OrgId: orgId, RoleId: roleId, Status: types.AcctStatus(status.Int64()), IsOrgAdmin: isAdmin}
+}
+
+// getter to get a org record from the contract
+func (p *PermissionCtrl) populateOrgToCache(orgId string) *types.OrgInfo {
+	permOrgInterface := &pbind.OrgManagerSession{
+		Contract: p.permOrg,
+		CallOpts: bind.CallOpts{
+			Pending: true,
+		},
+	}
+	org, parentOrgId, ultimateParentId, orgLevel, orgStatus, _ := permOrgInterface.GetOrgDetails(orgId)
+	if orgStatus.Int64() == 0 {
+		return nil
+	}
+	orgInfo := types.OrgInfo{OrgId: org, ParentOrgId: parentOrgId, UltimateParent: ultimateParentId, Status: types.OrgStatus(orgStatus.Int64()), Level: orgLevel}
+	// now need to build the list of sub orgs for this org
+	subOrgIndexes, _ := permOrgInterface.GetSubOrgIndexes(orgId)
+
+	if len(subOrgIndexes) == 0 {
+		return &orgInfo
+	}
+
+	// range through the sub org indexes and get the org ids to populate the suborg list
+	for _, s := range subOrgIndexes {
+		subOrgId, _, _, _, _, err := permOrgInterface.GetOrgInfo(s)
+		if err == nil {
+			orgInfo.SubOrgList = append(orgInfo.SubOrgList, orgId+"."+subOrgId)
+		}
+	}
+	return &orgInfo
+}
+
+// getter to get a role record from the contract
+func (p *PermissionCtrl) populateRoleToCache(roleKey *types.RoleKey) *types.RoleInfo {
+	permRoleInterface := &pbind.RoleManagerSession{
+		Contract: p.permRole,
+		CallOpts: bind.CallOpts{
+			Pending: true,
+		},
+	}
+	roleDetails, _ := permRoleInterface.GetRoleDetails(roleKey.RoleId, roleKey.OrgId)
+	if roleDetails.OrgId == "" {
+		return nil
+	}
+	return  &types.RoleInfo{OrgId: roleDetails.OrgId, RoleId: roleDetails.RoleId, IsVoter: roleDetails.Voter, IsAdmin: roleDetails.Admin, Access: types.AccessType(roleDetails.AccessType.Int64()), Active: roleDetails.Active}
+}
+// getter to get a role record from the contract
+func (p *PermissionCtrl) populateNodeCache(url string) *types.NodeInfo {
+	permNodeInterface := &pbind.NodeManagerSession{
+		Contract: p.permNode,
+		CallOpts: bind.CallOpts{
+			Pending: true,
+		},
+	}
+	nodeDetails, _ := permNodeInterface.GetNodeDetails(url)
+	if nodeDetails.NodeStatus.Int64() == 0{
+		return nil
+	}
+	return  &types.NodeInfo{OrgId:nodeDetails.OrgId, Url: nodeDetails.EnodeId, Status:types.NodeStatus(nodeDetails.NodeStatus.Int64()) }
+}
+
+// getter to get a node record from the contract
+func (p *PermissionCtrl) populateNodeCacheAndValidate(hexNodeId, ultimateParentId string) bool {
+	permNodeInterface := &pbind.NodeManagerSession{
+		Contract: p.permNode,
+		CallOpts: bind.CallOpts{
+			Pending: true,
+		},
+	}
+	txnAllowed := false
+	passedEnode, _ := enode.ParseV4(hexNodeId)
```

Comment:
handle error returned

Reply:
This is getting called from `ValidateNodeForTxn` where already `ParseV4` error is handled. Hence error handling here will be redundant.